### DISCUSSION TITLE
Can I construct a length from a float?

### DISCUSSION THREAD

**Discussion by ja-he** [typst/typst#4861] [General]
I really just want to construct a length from a float, e.g. say that the float `w` I have should be converted to `w`-many `em` or perhaps `w`-many `mm`, etc.

The obvious things I tried did not work (trying to construct with `length(w)`[^1] or similar).

Perhaps there is an obvious way and I didn't find[^2] it;
I found an issue discussing how to do math between lengths and floats and I agree with the conclusion that it doesn't make much sense.

---

As an aside, my use case was constructing a simple timeline where I say:

```typ
#schedule(events-ordered: (
  (name:"foo", mins:20, color: green),
  (name:"bar", mins:40, color: blue),
))
```

and it turns out a `grid` with a bunch of `rect`s, each sized proportional to the `event.mins / total-time`, or similar.

---

[^1]: I was really hoping for something like `length(w, unit: "em")`, to be honest
[^2]: I searched issues and discussions for something like `float length` and googled for stuff like `typst float to length`

**Comment by LaurenzV** [comment]
You can do something like this:
```typ
#{
  let num = 1
  let unit = num * 1em
  unit
}
```

Maybe there should be a more explicit way of doing that, though.

**Reply by ja-he** [reply]
that works well, thank you

**Comment by ja-he** [comment]
_edit: commented at the same time as answer above, which nullifies this comment_

<details>
<summary>original eureka-moment</summary>

Oh hey, I think I figured it out:
`eval(str(w) + "em")` works!

I guess my question then becomes: **Is there a better way?**
I would guess that the performance of gluing together a string and then parsing it again is much poorer than something like my wish-syntax above.
Not a real difference for normal documents but something that does this a lot would probably have a real performance detriment from `eval`-ing all over?

</details>

The answer to my ponderings there is just the answer above.

**Reply by LaurenzV** [reply]
AFAIK what I wrote above is the shortest way of doing it, currently. Basically just multiply the number by 1em/mm/cm etc.
